### PR TITLE
feat: add bigframes.bigquery.approx_top_count

### PR DESCRIPTION
**Pull request by chelsea-lin**
feat: add bigframes.bigquery.approx_top_count

- [X] Fixes internal issue 363004193 
- [X] Ensure the tests and linter pass
- [X] Code coverage does not decrease (if any source code was changed)
- [X] Appropriate docs were updated (if necessary)

Fixes internal issue 363004193 🦕

### CODE REVIEW

File: bigframes/bigquery/__init__.py [hunk 1]
```diff
@@ -272,6 +272,44 @@ def json_extract_array(
...
+    The number parameter specifies the number of elements returned.
+
+    Each `STRUCT` contains two fields. The first field (named `value`) contains an input
+    value. The second field (named `count`) contains an `INT64` specifying the number
+    of times the value was returned.
+
+    Returns `NULL` if there are zero input rows.
+
+    **Examples:**
+        >>> import bigframes.pandas as bpd
```
Comment (shobsi) [inline bigframes/bigquery/__init__.py:293]: I think we need newline after **Examples:** to render it nicely

Comment (chelsea-lin) [reply bigframes/bigquery/__init__.py:294]: Good catches! Fixed.

**Review by Genesis929 — APPROVED** [review]

File: tests/system/small/bigquery/test_approx_agg.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Google LLC
```
Comment (shobsi) [inline tests/system/small/bigquery/test_approx_agg.py:1]: Are these tests deterministic, given they are about an approx function?

Comment (chelsea-lin) [reply tests/system/small/bigquery/test_approx_agg.py:1]: I observed consistent results from local running. The consistency might be due to the small dataset.

**Pull request merged**